File: week2/read_file.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int main(void) {
    const char *filename = "/home/ug/halpenr4/data/week2/output_c.txt";
    char buffer[256];
    int line_count = 0;

    clock_t start_time = clock();

    FILE *fp = fopen(filename, "r");
    if (fp == NULL) {
        printf("Error: could not open file for reading.\n");
        return 1;
    }

    while (fgets(buffer, sizeof(buffer), fp) != NULL) {
        line_count++;
    }

    fclose(fp);

    clock_t end_time = clock();
    double run_time = (double)(end_time - start_time) / CLOCKS_PER_SEC;

    printf("Read %d lines from %s\n", line_count, filename);
    printf("Runtime: %.6f seconds\n", run_time);

    return 0;
}

File: week2/read_file.py
import time
import os

filename = os.path.expanduser("~/data/week2/output.txt")

start_time = time.time()

with open(filename, "r") as f:
    data = f.readlines()

end_time = time.time()

run_time = end_time - start_time

print(f"Read {len(data)} lines from {filename}")
print(f"Runtime: {run_time:.6f} seconds")
